### PR TITLE
Fix regressions in Billboard properties

### PR DESCRIPTION
**Pull request by ptrgags**
Fixes some regressions that were causing the [Clamp to Terrain Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html?src=Clamp%20to%20Terrain.html&label=All) to break when you select "Draw Billboard" 

There weren't any specs for Billboard, though hm... I should check BillboardCollectionSpec too.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ptrgags!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/fix-billboard-depth-test-dist/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

File: Specs/Scene/BillboardCollectionSpec.js [hunk 1]
```diff
@@ -241,6 +241,170 @@ describe(
...
+        b.distanceDisplayCondition = 10;
+      }).toThrowDeveloperError();
+      expect(function () {
+        b.disableDepthTestDistance = "far";
+      }).toThrowDeveloperError();
+    });
+
+    it("image property setter creates image with GUID for non-uris", function () {
+      var b = billboards.add();
+      b.image = 42;
```
Comment (ptrgags) [inline Specs/Scene/BillboardCollectionSpec.js:402]: @lilleyse this setter is weird, if the value is defined and isn't a String, Resource, Image, etc, it creates a new image with a GUID... how is this supposed to be used?

**Comment by ptrgags** [comment]
@lilleyse updated!

File: Specs/Scene/BillboardCollectionSpec.js (hunk 1 above)
Comment (lilleyse) [reply Specs/Scene/BillboardCollectionSpec.js:402]: I'm not completely sure, but my best guess is it has something to do with `LabelCollection`

**Comment by lilleyse** [comment]
Looks good, will merge when CI passes

**Pull request merged**